app/models/foreman_tasks/task/dynflow_task.rb

```diff
@@ -102,10 +104,16 @@ def self.consistency_check
             logger.warn("Task %s updated at consistency check: %s" % [task.id, changes.inspect])
           end
         rescue => e
-          Foreman::Logging.exception("Failed at consistency check for task #{task.id}", e, :logger => logger)
+          Foreman::Logging.exception("Failed at consistency check for task #{task.id}", e, :logger => 'foreman-tasks')
```

Comment:
Or I would be ok with the same PR, just keeping the change in separate commit and pairing it with a separate issue